sys/sys/types.h

```diff
@@ -251,6 +251,12 @@ struct cap_rights;
 typedef	struct cap_rights	cap_rights_t;
 #endif
 
+#ifdef __CHERI_PURE_CAPABILITY__
+typedef __uintptr_t	vm_pointer_t;
+#else
+typedef __vm_offset_t	vm_pointer_t;
+#endif
+
 /*
  * Types suitable for exporting physical addresses, virtual addresses
  * (pointers), and memory object sizes from the kernel independent of native
```

Comment:
This block comment could do with updating at some point... :)

Reply:
It's actually mostly still true for the things it covers.  It is used for debugging interfaces like kvm(3) which will only ever use virtual addresses and not actual cap pointers.

However, I do think looking at this in context that we should move vm_pointer_t down next to `vm_offset_t`.

Reply:
Do we want a __vm_pointer_t in sys/sys/_types.h?

Reply:
I agree about moving it next to `vm_offset_t`.

Reply:
Hmmmmm.  I ended up adding a `__uintcap_t` so we could use it in headers exported to userland for things like `struct kevent`.  I think in this case we can get away without needing `__vm_pointer_t` since we don't intend on exporting it to userland.